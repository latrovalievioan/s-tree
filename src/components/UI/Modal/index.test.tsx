import { render, screen, cleanup, fireEvent } from '@testing-library/react';
import { Modal } from '.';

describe('Modal', () => {
  let count = 0;
  beforeEach(() => {
    render(
      <Modal
        title="test"
        onClose={() => {
          count++;
        }}
        nonEscapable={false}
      >
        <div>TEST</div>
      </Modal>
    );
  });

  afterEach(() => {
    cleanup();
  });

  it("Should render the component and it shouldn't be visible by default", () => {
    const modal = screen.getByTestId('modal-test');

    expect(modal).toBeInTheDocument();
    expect(modal).not.toBeVisible();
  });

  it('Should render the passed children correctly', () => {
    const modal = screen.getByTestId('modal-test');

    expect(modal).toContainHTML('<div>TEST</div>');
  });

  it('Should execute the onClose prop when escapable and the Close icon is clicked', () => {
    const button = screen.getByTestId('icon-button');

    fireEvent.click(button);

    expect(count).toEqual(1);
  });

  it('Matches snapshot', () => {
    const modal = screen.getByTestId('modal-test');

    expect(modal).toMatchSnapshot();
  });
});
