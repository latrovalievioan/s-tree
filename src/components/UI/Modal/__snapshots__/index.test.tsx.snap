// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Modal > Matches snapshot 1`] = `
<dialog
  data-testid="modal-test"
>
  <header>
    <h2>
      test
    </h2>
    <button
      class="action"
      data-testid="icon-button"
      title="Close"
    >
      <svg
        aria-hidden="true"
        fill="none"
        height="24"
        viewBox="0 0 20 20"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m13 7-6 6m0-6 6 6m6-3a9 9 0 1 1-18 0 9 9 0 0 1 18 0Z"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </button>
  </header>
  <div>
    TEST
  </div>
</dialog>
`;
